src/frequenz/sdk/actor/power_distributing/power_distributing.py

```diff
@@ -62,6 +64,41 @@ class _User:
     """The bidirectional channel to communicate with the user."""
 
 
+@dataclass
+class _CacheEntry:
+    """Represents an entry in the cache with expiry time."""
+
+    inv_bat_pair: InvBatPair
+    """The inverter and adjacent battery data pair."""
+
+    expiry_time: int
+    """The expiration time (taken from the monotonic clock) of the cache entry."""
+
+    @classmethod
+    def from_ttl(
+        cls, inv_bat_pair: InvBatPair, ttl: timedelta = timedelta(hours=2.5)
+    ) -> Self:  # type: ignore[name-defined] # pylint: disable=undefined-variable
+        """Initialize a CacheEntry instance from a TTL (Time-To-Live).
+
+        Args:
+            inv_bat_pair: the inverter and adjacent battery data pair to cache.
+            ttl: the time a cache entry is kept alive.
+
+        Returns:
+            this class instance.
+        """
+        return cls(inv_bat_pair, time.monotonic_ns() + int(ttl.total_seconds() * 1e9))
+
+    @property
```

Comment:
Ah, sorry, I know I suggested this but didn't realize that this function takes the time, I think it was a bad idea to make a property something that depends on the clock, because `entry.has_expired` can have different values depending on when is called even if there were no modifications to the instance. Again, sorry, my bad, but I think it is better to keep it as a regular method.

Reply:
no problem, I also missed that. I'll quickly update it

Reply:
done